server/services/cards.js

```diff
@@ -5,15 +5,17 @@ module.exports.assign = (users, number_of_cards) => {
     for (let j = 0; j < number_of_cards; j++) {
         const min = 1;
         const max = 50;
-        const random = Math.floor(Math.random() * (max - min) + min);
-        const availableGlobally = allCards.indexOf(random) < 0;
-        const inCurrentHand = cardsSet.indexOf(random) >= 0;
-        while (!availableGlobally || inCurrentHand) {
-            const random = Math.floor(Math.random() * (max - min) + min);
-            const available = allCards.indexOf(random) < 0;
-            const inCurrentHand = cardsSet.indexOf(random) >= 0;
+        let random = Math.floor(Math.random() * (max - min) + min);
+        
+        let availableGlobally = allCards.some(card => card.id !== random);
+        let inCurrentHand = cardsSet.some(card => card.id === random);
+        
+        while (availableGlobally || inCurrentHand) {            
+            random = Math.floor(Math.random() * (max - min) + min);
+            availableGlobally = allCards.some(card => card.id === random);
+            inCurrentHand = cardsSet.some(card => card.id === random);
         }
-        cardsSet.push(random);
+        cardsSet.push({ id: random, played: false });
```

Comment:
Are you using this `played` attribute on the front end? Because I don't think we need it on the back end, as we keep a list of played cards